priam/src/main/java/com/netflix/priam/backup/AbstractFileSystem.java

```diff
@@ -257,6 +258,16 @@ public boolean checkObjectExists(Path remotePath) {
         return remoteFileExist;
     }
 
+    @Override
+    public void deleteRemoteFiles(List<Path> remotePaths) throws BackupRestoreException {
+        if (remotePaths == null) return;
+
+        deleteFiles(remotePaths);
+        objectCache.invalidateAll(remotePaths);
```

Comment:
I would invalidate cache before deleting remote files. Also, code documentation on why this is done this way.